src/main/java/uk/gov/hmcts/darts/notification/service/impl/NotificationServiceImpl.java

```diff
@@ -1,38 +1,123 @@
+        String[] emailAddressList = emailAddresses.split(",");
+        for (String emailAddress : emailAddressList) {
+            saveNotificationToDb(request.getEventId(), request.getCaseId(), StringUtils.trim(emailAddress), request.getTemplateValues());
+        }
+    }
+
+    private Notification saveNotificationToDb(String eventId, String caseId, String emailAddress, String templateValues) {
+        if (!emailValidator.isValid(emailAddress)) {
+            log.warn("The supplied email address, {}, is not valid, and so has been ignored.", emailAddress);
+            return null;
+        }
         Notification dbNotification = new Notification();
-        dbNotification.setEventId(createNotificationReq.getEventId());
-        dbNotification.setCaseId(createNotificationReq.getCaseId());
-        dbNotification.setEmailAddress(createNotificationReq.getEmailAddress());
+        dbNotification.setEventId(eventId);
+        dbNotification.setCaseId(caseId);
+        dbNotification.setEmailAddress(emailAddress);
         dbNotification.setStatus(String.valueOf(NotificationStatus.OPEN));
         dbNotification.setAttempts(0);
-        dbNotification.setTemplateValues(createNotificationReq.getTemplateValues());
-        Timestamp now = new Timestamp(System.currentTimeMillis());
-        dbNotification.setCreatedDateTime(now);
-        dbNotification.setLastUpdatedDateTime(now);
+        dbNotification.setTemplateValues(templateValues);
 
         return notificationRepo.saveAndFlush(dbNotification);
+    }
+
+
+    @Override
+    public void sendNotificationToGovNotify() {
 
+        List<Notification> notificationEntries = notificationRepo.findByStatusIn(STATUS_ELIGIBLE_TO_SEND);
+        for (Notification notification : notificationEntries) {
+            String templateId;
+            try {
+                templateId = templateIdHelper.findTemplateId(notification.getEventId());
+            } catch (TemplateNotFoundException e) {
+                markNotificationAsFailed(notification);
+                break;
+            }
+
+            GovNotifyRequest govNotifyRequest = null;
+            try {
+                govNotifyRequest = GovNotifyRequestMapper.map(notification, templateId);
+                govNotifyService.sendNotification(govNotifyRequest);
+                markNotificationAsSent(notification);
+            } catch (JsonProcessingException e) {
+                markNotificationAsFailed(notification);
+            } catch (NotificationClientException e) {
+                log.error(
+                    "GovNotify has responded back with an error while trying to send Notification Id {}. Request={}, error={}",
+                    notification.getId(),
+                    govNotifyRequest,
+                    e.getMessage()
+                );
+                incrementNotificationFailureCount(notification);
+            }
+        }
+    }
+
+    private void markNotificationAsFailed(Notification notification) {
```

Comment:
This could just be a  'updateNotificationStatus' method and pass in a 'NotificationStatus' rather than having a method for each one?

Reply:
done